src/codecoverage/bot/code_coverage_bot/hgmo.py

```diff
@@ -32,10 +34,46 @@ def __enter__(self):
 
         logger.info('hgmo is running', pid=self.__get_pid())
 
-        return 'http://localhost:8000'
+        return self
 
     def __exit__(self, type, value, traceback):
         pid = self.__get_pid()
         os.killpg(os.getpgid(pid), signal.SIGTERM)
         os.remove(self.pid_file)
         logger.info('hgmo has been killed')
+
+    def get_pushes(self, startID=None, changeset=None):
+        assert startID is not None or changeset is not None
+
+        params = {
+            'version': 2,
+            'full': 1
+        }
+
+        if startID is not None:
+            params['startID'] = startID
+
+        if changeset is not None:
+            params['changeset'] = changeset
+
+        r = requests.get('http://localhost:8000/json-pushes', params=params)
+
+        r.raise_for_status()
+        return r.json()
+
+    def get_push_changesets(self, changeset):
+        push_data = self.get_pushes(changeset=changeset)
+
+        return sum((data['changesets'] for data in push_data['pushes'].values()), [])
```

Comment:
I find this pretty hard to read. I thought at first that the function would output a number (because of `sum`), but having the `start=[]` makes it behave as a reduce operation.

That's smart but i would add a comment for dumb devs like myself.